modules/lib/lib-event/src/main/resources/lib/xp/event.ts

```diff
@@ -0,0 +1,136 @@
+import * as Lib from './event';
+
+/*
+ * Global library definitions for require and __non_webpack_import__
+ */
+
+type LibType = typeof Lib;
+
+declare global {
+    interface XpEventLibrary extends LibType {}
+}
+
+
+export interface ListenerParams<EventData extends object = EnonicEventData> {
+    /**
+     * Event type pattern
+     */
+    type: EnonicEventTypes;
+
+    /**
+     * Callback event listener
+     */
+    callback: (event: EnonicEvent<EventData>) => void;
+
+    /**
+     * Local events only (default to false)
+     */
+    localOnly?: boolean;
+}
+
+export interface SendParams {
+    /**
+     * Event type
+     */
+    type: EnonicEventTypes;
+
+    /**
+     * `true` if it should be distributed in cluster
+     */
+    distributed?: boolean;
+
+    /**
+     * Additional data for event.
+     */
+    data?: object;
+}
+
+export interface EnonicEvent<EventData extends object = EnonicEventData> {
+    readonly type: EnonicEventTypes | string;
+    readonly timestamp: number;
+    readonly localOrigin: boolean;
+    readonly distributed: boolean;
+    readonly data: EventData;
+}
+
+export interface EnonicEventData {
+    readonly nodes: ReadonlyArray<EnonicEventDataNode>;
+    readonly state?: string; // event type <'node.stateUpdated'>
+}
+
+export interface EnonicEventDataNode {
+    readonly id: string;
+    readonly path: string;
+    readonly branch: string;
+    readonly repo: string;
+    readonly newPath?: string; // event type <'node.moved' | 'node.renamed'>
+}
+
+type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);
```

Comment:
This is used for improving code completion right? You should add a comment to this type to document it. It can be difficult to understand the usage of the type.

Reply:
Yup, it's for the completion. Since we expand the type with `string`, that will lead the editors just omit the more narrow types, if just the `|` operator is used.
Will add a comment. I'm just not sure if it'll stay here or be moved to other place, so that such tricks can be used across other libraries.
But you are right, adding at least `// TODO` will be a good thing so I don't forget to properly comment it later.